numpy/core/arrayprint.py

```diff
@@ -521,6 +509,8 @@ def array2string(a, max_line_width=None, precision=None,
     if formatter is None:
         formatter = _formatter
 
+    if style is not None and a.shape == () and a.dtype.names is None:
+        return style(a[()])
```

Comment:
I'd argue this should be all the way upstream in `ndarray.__str__` - ie, complete ignore `np.setprintfunction` for the 0d string case.